source/common/http/codec_client.h

```diff
@@ -219,6 +220,9 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
   void onEvent(Network::ConnectionEvent event) override;
   // Pass watermark events from the connection on to the codec which will pass it to the underlying
   // streams.
+  void onAboveWriteBufferOverflowWatermark() override {
```

Comment:
Do we need to pass this on, or do we close the connection?
Arguably it's cleaner to reset streams then close the connection but if we think this is abuse we may have a case for force-close

Reply:
Yeah, I think you're right, that we shouldn't pass the overflow on. Closing the connection is the simplest thing to do.